environment/src/main/java/com/sequenceiq/environment/environment/encryption/EnvironmentEncryptionService.java

```diff
@@ -0,0 +1,84 @@
+import com.sequenceiq.cloudbreak.cloud.EncryptionResources;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.CloudPlatformVariant;
+import com.sequenceiq.cloudbreak.cloud.model.Platform;
+import com.sequenceiq.cloudbreak.cloud.model.Region;
+import com.sequenceiq.cloudbreak.cloud.model.Variant;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.CreatedDiskEncryptionSet;
+import com.sequenceiq.cloudbreak.cloud.model.encryption.DiskEncryptionSetCreationRequest;
+import com.sequenceiq.environment.credential.v1.converter.CredentialToCloudCredentialConverter;
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.service.EnvironmentTagProvider;
+import com.sequenceiq.environment.parameter.dto.AzureParametersDto;
+import com.sequenceiq.environment.parameter.dto.AzureResourceGroupDto;
+import com.sequenceiq.environment.parameter.dto.ParametersDto;
+import com.sequenceiq.environment.parameter.dto.ResourceGroupUsagePattern;
+
+@Component
+public class EnvironmentEncryptionService {
+
+    private final CredentialToCloudCredentialConverter credentialToCloudCredentialConverter;
+
+    private final CloudPlatformConnectors cloudPlatformConnectors;
+
+    private final EnvironmentTagProvider environmentTagProvider;
+
+    public EnvironmentEncryptionService(CredentialToCloudCredentialConverter credentialToCloudCredentialConverter,
+            CloudPlatformConnectors cloudPlatformConnectors, EnvironmentTagProvider environmentTagProvider) {
+        this.credentialToCloudCredentialConverter = credentialToCloudCredentialConverter;
+        this.cloudPlatformConnectors = cloudPlatformConnectors;
+        this.environmentTagProvider = environmentTagProvider;
+    }
+
+    public CreatedDiskEncryptionSet createEncryptionResources(EnvironmentDto environmentDto, Environment environment) {
+        EncryptionResources encryptionResources = getEncryptionResources(environment.getCloudPlatform());
+        return encryptionResources.createDiskEncryptionSet(createEncryptionResourcesCreationRequest(environmentDto));
+    }
+
+    public EncryptionResources getEncryptionResources(String cloudPlatform) {
+        CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+        return Optional.ofNullable(cloudPlatformConnectors.get(cloudPlatformVariant))
+                .map(CloudConnector<Object>::encryptionResources)
+                .orElseThrow(() -> new EncryptionResourcesNotFoundException("No Encryption resources component found for cloud platform: " + cloudPlatform));
+    }
+
+    public DiskEncryptionSetCreationRequest createEncryptionResourcesCreationRequest(EnvironmentDto environment) {
```

Comment:
Like for `getEncryptionResources()`, this should be either `private` or package private, and bear the `@VisibleForTesting` annotation in the latter case.